orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/ToggleablePauseTask.java

```diff
@@ -0,0 +1,58 @@
+package com.netflix.spinnaker.orca.pipeline.tasks;
+
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService;
+import com.netflix.spinnaker.orca.api.pipeline.RetryableTask;
+import com.netflix.spinnaker.orca.api.pipeline.TaskResult;
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+import java.time.Duration;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * A task that just waits indefinitely based on a feature toggle specified in the `pauseToggle`
+ * property of the stage context. Useful for pausing stages as part of migrations, for instance.
+ */
+@Component
+public class ToggleablePauseTask implements RetryableTask {
+  private final Logger log = LoggerFactory.getLogger(getClass());
+  private DynamicConfigService dynamicConfigService;
+
+  @Autowired
+  public ToggleablePauseTask(DynamicConfigService dynamicConfigService) {
+    this.dynamicConfigService = dynamicConfigService;
+  }
+
+  @Override
+  @Nonnull
+  public TaskResult execute(@Nonnull final StageExecution stage) {
+    final String pauseToggle =
+        stage.getContext().containsKey("pauseToggle")
+            ? stage.getContext().get("pauseToggle").toString()
+            : null;
+
+    if (pauseToggle != null && dynamicConfigService.isEnabled(pauseToggle, false)) {
+      log.debug(
+          "{} stage currently paused based on {} toggle. Waiting...", stage.getName(), pauseToggle);
+      return TaskResult.RUNNING;
+    } else {
+      log.debug(
+          "{} stage currently unpaused based on {} toggle. Carrying on...",
+          stage.getName(),
+          pauseToggle);
+      return TaskResult.SUCCEEDED;
+    }
+  }
+
+  @Override
+  public long getBackoffPeriod() {
+    return Duration.ofMinutes(1).toMillis();
+  }
+
+  @Override
+  public long getTimeout() {
+    return Long.MAX_VALUE;
```

Comment:
i would probably bound this somewhat. Maybe a few hours? 4 (can also make it configurable)?

Reply:
Is this in milliseconds?

Reply:
Also, I think I disagree with this in principle. The whole point of this task is to run indefinitely until the toggle has been turned off. Adding a timeout would defeat the purpose. The task can still be cancelled if someone got tired of waiting.

Reply:
I hear what you're saying. It's just weird to have waits that are unbounded. We could end up with a bunch of executions that are waiting indefinitely and we have to cancel them all manually. While I understand it's not really possible (or easily recoverable) with your current implementation. Someone else who might find this task useful and want to use may not realize it/have a bug.
I am ok not bounding this for the given use case, but the idea that there are no guard rails in the "default" case is uncomfortable to me

Reply:
OK. I'll add a 24 hour timeout as a compromise.